contracts/operators/CollectionBatchBuyOperator.sol

```diff
@@ -147,38 +164,36 @@ contract CollectionBatchBuyOperator is IOperator {
             }
 
             {
-                uint96 ethUsed;
+                uint96 receivedTokensValue = 0;
                 for (uint256 j; j < call.tokensToBuy.length; ++j) {
                     uint256 tokenId = call.tokensToBuy[j].tokenId;
 
-                    // Check whether the NFT was successfully bought.
                     if (op.nftContract.safeOwnerOf(tokenId) == receiver) {
-                        ethUsed += call.tokensToBuy[j].price;
-                        ++tokensBought;
+                        purchasedTokenIds[numTokensBought++] = tokenId;
+                        receivedTokensValue += call.tokensToBuy[j].price;
 
-                        // Add the token to the list of tokens to finalize.
-                        tokenIds[tokensBought - 1] = tokenId;
+                        if (!ex.isReceivedDirectly) {
+                            // Transfer the NFT to the Party.
+                            op.nftContract.safeTransferFrom(address(this), msg.sender, tokenId);
```

Comment:
Upon second look, I realize we don't actually need to do `safeTransferFrom` here. We know the Party can receive NFTs

Reply:
good call